src/algebra/group/basic.lean

```diff
@@ -8,6 +8,58 @@ import logic.function.basic
 
 universe u
 
+section associative
+variables {α : Type u} (f : α → α → α) [is_associative α f] (x y : α)
+
+/--
+Composing two associative operations of `f : α → α → α` on the left
+is equal to an associative operation on the left.
+-/
+@[simp] lemma comp_assoc_left : (f x) ∘ (f y) = (f (f x y)) :=
```

Comment:
Usually assoc rules are not added to the default simp-set. Is there a good reason to make these simp-by-default?

Reply:
Doesn't this suffer from the same problem as attempting to use `is_monoid_hom` in simp rules?

Reply:
You're probably right here. The lemmas below that, that prove the exact statement, but for groups and semigroups, could have been proved with `by simp`, but even the linter doesn't complain that these are duplicates in the `simp` tree. So setting these as simp-lemmas did work (or can work), but the real power in simp-lemmas comes from the semigroup and add_semigroup lemmas, which is where most of where associate operations would come from.